Extension/src/LanguageServer/client.ts

```diff
@@ -405,23 +412,63 @@ class DefaultClient implements Client {
         }
     }
 
-    public onDidChangeCustomConfigurations(provider: CustomConfigurationProvider): void {
-        let documentUris: vscode.Uri[] = [];
-        this.trackedDocuments.forEach(document => documentUris.push(document.uri));
+    public onRegisterCustomConfigurationProvider(provider: CustomConfigurationProvider): Thenable<void> {
+        return this.notifyWhenReady(() => {
+            let selectedProvider: string = this.configuration.CurrentConfiguration.configurationProvider;
+            if (!selectedProvider) {
+                let ask: PersistentFolderState<boolean> = new PersistentFolderState<boolean>("Client.registerProvider", true, this.RootPath);
+                if (ask.Value) {
+                    const message: string = `${provider.name} would like to provide IntelliSense configuration information for this workspace`;
```

Comment:
Should there be a period?

Reply:
Added a period

---
In reply to: [196529952](https://github.com/Microsoft/vscode-cpptools/pull/2145#discussion_r196529952) [](ancestors = 196529952)